### PR TITLE
feat(`effect/Micro)`: add isMicroCause

### PR DESCRIPTION
**Pull request by juliusmarminge**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds guard for `MicroCause` so that you can check the value before passing it to `Micro.causeSquash`

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: packages/effect/src/Micro.ts [hunk 1]
```diff
@@ -171,6 +171,13 @@ export type MicroCauseTypeId = typeof MicroCauseTypeId
  */
 export type MicroCause<E> = MicroCause.Die | MicroCause.Fail<E> | MicroCause.Interrupt
 
+/**
+ * @since 3.7.0
+ * @experimental
+ * @category guards
+ */
+export const isMicroCause = <E>(self: unknown): self is MicroCause<E> => hasProperty(self, MicroCauseTypeId)
```
Comment (juliusmarminge) [inline packages/effect/src/Micro.ts:179]: Maybe this is more correct:

```suggestion
export const isMicroCause = (self: unknown): self is MicroCause<unknown> => hasProperty(self, MicroCauseTypeId)
```

or

```suggestion
export const isMicroCause = <E = unknown>(self: unknown): self is MicroCause<E> => hasProperty(self, MicroCauseTypeId)
```

File: .changeset/chilled-toys-share.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"effect": minor
```
Comment (tim-smart) [inline .changeset/chilled-toys-share.md:2]:
```suggestion
"effect": patch
```

We can do patch releases for modules tagged with experimental.

File: packages/effect/src/Micro.ts [hunk 3]
```diff
@@ -171,6 +171,13 @@ export type MicroCauseTypeId = typeof MicroCauseTypeId
  */
 export type MicroCause<E> = MicroCause.Die | MicroCause.Fail<E> | MicroCause.Interrupt
 
+/**
+ * @since 3.7.0
```
Comment (tim-smart) [inline packages/effect/src/Micro.ts:175]:
```suggestion
 * @since 3.6.6
```

**Comment by juliusmarminge** [comment]
it seems like this still got merged into `next-minor`

**Comment by tim-smart** [comment]
> it seems like this still got merged into `next-minor`

Ah yep, forgot to change the base branch.

Minors get released every two weeks or so, will have to wait sorry.

**Pull request merged**